torch/csrc/jit/codegen/cuda/scheduler_registry.cpp

```diff
@@ -37,26 +63,35 @@ class SingleReductionScheduler : public SchedulerEntry {
     getHeuristics(fusion, ee);
   }
 
+  //! Check if the reduction heuristics apply in given fusion
   static bool canSchedule(Fusion* fusion) {
-    int reduction_count = 0;
-    for (auto expr : fusion->exprs()) {
-      if (auto red = expr->as<ReductionOp>()) {
-        // Check trivial
-        if (!isTrivialReduction(red)) {
-          reduction_count++;
-          if (reduction_count > 1) {
-            return false;
-          }
+    auto red_ops = findReductionOps(fusion);
+    if (red_ops.size() != 1) {
+      return false;
+    }
+
+    auto red_tv = red_ops[0]->out()->as<TensorView>();
+
+    // Not allowing broadcasting reduction result to support
+    //  grid reduction. This is an overkill might want to consider
+    //  trying to get the heuristics and check only if grid reduction is
+    //  required.
+    auto uses = DependencyCheck::getAllUseChains(red_tv);
+    for (auto& chain : uses) {
+      for (auto val : chain) {
+        if (val->definition()->isA<BroadcastOp>()) {
+          return false;
```

Comment:
If the value is an output with no other dependencies, then the reduction scheduler is valid. Otherwise return false.

Reply:
Added this logic. Thanks! Still leaves the cases where the broadcast has dependencies which don't resolve the broadcast. But I guess current logic captures enough interesting use cases and we can leave the rest for follow ups.